### ISSUE TITLE
Integration Tests

### ISSUE THREAD

**Issue by rhiananthony** [HumanCellAtlas/dcp#81]
https://docs.google.com/document/d/17tvOujycch6VXRmYWh6LYo1MaZyf2YuYEyB_-9LrrII/edit#heading=h.2rlu4knkzel9

Other ACs:
- Key thing is to have a framework that everyone can add tests to
- Need to make time to add tests to integration testing suite
- Find data, make a spreadsheet - can wranglers help?
- Test log itself needs to be stored somewhere

**Comment by rhiananthony** [comment]
Kylee - May 22, 2018 - 11:21AM
After #48 we will need to implement integration tests.
The goal is to be able to confidently use automated deployment from integration->staging after we implement these tests.

Kylee - May 2, 2018 - 7:11PM
I believe Hannes is working on this hopefully in time for the May f2f

Rex - Apr 30, 2018 - 5:47PM
Started to draft a doc for this: https://docs.google.com/document/d/1cFKXAGeKee4uR5HNXJgLy8sPpJLN1gSVKyJpedsG0U0/edit#heading=h.xgjl2srtytjt

**Comment by hannes-ucsc** [comment]
We do have an inter-box integration test. So what more needs to be added?

**Comment by kbergin** [comment]
A [comment](https://humancellatlas.slack.com/archives/C75G0R97E/p1537890634000100) from @kislyuk in slack on 9/25 on integration tests that I wanted to document somewhere. 
>For all but the lowest level changes, my suggestion is to have a DCP-wide PR-based integration test environment. i.e. for any PR, the test would redeploy all components fully automatically from an integration branch, except for the PR component (which would deploy from the PR branch) and the integration test would run. We would want to make the deploy/integration process as fast as possible for this, and tests would need to be serialized not to step on each other’s toes. But that would give us a platform to quickly detect issues without forcing rollbacks

**Comment by sampierson** [comment]
This is a "nice to have".  I'm changing the title from "Integration Tests", which we have, to "PR-based integration tests with automatic deployment" and am ice-boxing it.

**Comment by brianraymor** [comment]
I'm not sure that "nice to have" meets the bar for _Icebox_? This is ~8 months old. @kislyuk - are there any plans to develop or pursue ideas further?
